src/Components/Locator/Locator.js

```diff
@@ -0,0 +1,72 @@
+import Button from '@material-ui/core/Button';
+import React from 'react';
+
+import { getLocation, getWeatherFromLocation } from '../../location/getLocation';
+
+export class Locator extends React.Component{
+  constructor(props){
+    super(props);
+    this.handleDebugClick = this.handleDebugClick.bind(this);
+    this.state = {
+      canLoad: false,
+    }
+  }
+
+  // Executed upon component appearing in app.
+  componentDidMount() {
+    this.runLocator();
+  }
+
+  async runLocator() {
+    let result;
+    result = await getLocation();
+    if (result.success) {
+      console.log(result);
+    }
+    else {
+      // Default to getting weather from Corvallis if the user doesn't consent to giving their location
+      result = await getWeatherFromLocation({
+        coords: {
+          latitude: 44.566667,
+          longitude: -123.283333
+        }
+      })
+      console.log("DEBUG: Could not find your location. (or wasn't given permission)");
```

Comment:
ultimately, should this be some type of alert displayed to the user so they know why their weather isn't loading?

Reply:
Yeah, that's definitely something we'll want. I've got it so an error banner appears if the weather data can't be retrieved though. (like an invalid API key or something)